From 932e5d4a193bfc97a89fe5647f52a53d3abb3c91 Mon Sep 17 00:00:00 2001
From: Frank Ramsay <framsay@redhat.com>
Date: Thu, 13 Nov 2014 18:29:56 -0500
Subject: [x86] uv_bau: Increase maximum CPUs per socket/hub

Message-id: <1415903397-11300-2-git-send-email-framsay@redhat.com>
Patchwork-id: 99959
O-Subject: [RHEL7.1 PATCH 1/2] x86: UV BAU: Increase maximum CPUs per socket/hub
Bugzilla: 1161183
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>

Bugzilla: 1161183:
Upstream Status: Submitted, looks like it will go though:
http://git.kernel.org/cgit/linux/kernel/git/tip/tip.git/commit/?id=3ab0c49fd6b1e80438d0a649584d292880112262
Build Info: Brew Build 8235386
Tested: tested on SGI lab UV system with 4x 18 core processors w/hyperthreading

Commit: 3ab0c49fd6b1e80438d0a649584d292880112262
Author: James Custer <jcuster@sgi.com>
Date: 2014-11-02 18:16:38 (GMT)

 We have encountered hardware with 18 cores/socket that gives 36 CPUs/socket
 with hyperthreading enabled. This exceeds the current MAX_CPUS_PER_SOCKET
 causing a failure in get_cpu_topology. Increase MAX_CPUS_PER_SOCKET to 64
 and MAX_CPUS_PER_UVHUB to 128.

 Signed-off-by: James Custer <jcuster@sgi.com>
 Cc: Russ Anderson <rja@sgi.com>
 Link: http://lkml.kernel.org/r/1414952199-185319-1-git-send-email-jcuster@sgi.com
 Signed-off-by: Thomas Gleixner <tglx@linutronix.de>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/include/asm/uv/uv_bau.h b/arch/x86/include/asm/uv/uv_bau.h
index 0b46ef2..ea3d3c1 100644
--- a/arch/x86/include/asm/uv/uv_bau.h
+++ b/arch/x86/include/asm/uv/uv_bau.h
@@ -33,8 +33,8 @@
  * Each of the descriptors is 64 bytes in size (8*64 = 512 bytes in a set).
  */
 
-#define MAX_CPUS_PER_UVHUB  64
-#define MAX_CPUS_PER_SOCKET  32
+#define MAX_CPUS_PER_UVHUB  128
+#define MAX_CPUS_PER_SOCKET  64
 #define ADP_SZ    64 /* hardware-provided max. */
 #define UV_CPUS_PER_AS   32 /* hardware-provided max. */
 #define ITEMS_PER_DESC   8
-- 
1.7.1